cost-model/src/cost_tracker.rs

```diff
@@ -122,10 +131,13 @@ impl CostTracker {
             .saturating_sub(in_flight_transaction_count);
     }
 
-    pub fn try_add(&mut self, tx_cost: &TransactionCost) -> Result<u64, CostTrackerError> {
+    pub fn try_add(&mut self, tx_cost: &TransactionCost) -> Result<UpdatedCosts, CostTrackerError> {
```

Comment:
for the use-case of forwarding, I don't see why we should/want to change this interface. Because there, we will always just add, no try about it.

Reply:
There is CU limits in Forwarder, so it is still `try_add` there. 

Thinking the change of signature is for a more generic purpose - to return the impact of successfully adding a transaction to cost tracker, it'd take more than just updated block_cost; The costliest account it updated can be useful.

Reply:
oh I see now. the limits in forwarder would be 100x for 100 batches. we modulo after try_add to see which batch it goes in.

i was thinking just always add then modulo at end, if modulo less than 100 put it in.
your way is much smarter